### ISSUE TITLE
tox invocation on Windows fails

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8578]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8578|
|Type|defect|
|Created|2016-07-09 09:49:20Z|
|Branch|https://github.com/twisted/twisted/tree/8578-rodrigc-tox-win32|

I tried to invoke tox on Windows and it failed:

```
C:\Users\Craig\twisted>tox -e py27-tests-windows
GLOB sdist-make: C:\Users\Craig\twisted\setup.py
py27-tests-windows inst-nodeps: C:\Users\Craig\twisted\build\dist\Twisted-16.2.0.zip
py27-tests-windows installed: appdirs==1.4.0,attrs==16.0.0,cffi==1.7.0,cryptography==1.4,defusedxml==0.4.1,docutils==0.12,e
num34==1.1.6,extras==1.0.0,fixtures==3.0.0,h2==2.4.0,hpack==2.2.0,hyperframe==4.0.1,idna==2.1,ipaddress==1.0.16,linecache2=
=1.0.0,pbr==1.10.0,priority==1.1.1,pyasn1==0.1.9,pyasn1-modules==0.0.8,pycparser==2.14,pycrypto==2.6.1,pyOpenSSL==16.0.0,py
piwin32==219,pyserial==3.1.1,python-mimeparse==1.5.2,python-subunit==1.2.0,service-identity==16.0.0,six==1.10.0,SOAPpy==0.1
2.22,testtools==2.2.0,traceback2==1.4.0,Twisted==16.2.0,unittest2==1.1.0,wstools==0.4.3
py27-tests-windows runtests: PYTHONHASHSEED='710'
py27-tests-windows runtests: commands[0] | C:\Users\Craig\twisted\build\py27-tests-windows\Scripts/trial --reactor=default
--reporter=verbose twisted
ERROR: invocation failed (errno 2), args: ['C:\\Users\\Craig\\twisted\\build\\py27-tests-windows\\Scripts/trial', '--reacto
r=default', '--reporter=verbose', 'twisted'], cwd: C:\Users\Craig\twisted\build\py27-tests-windows\tmp
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\tox.exe\__main__.py", line 9, in <module>
  File "c:\python27\lib\site-packages\tox\session.py", line 39, in main
    retcode = Session(config).runcommand()
  File "c:\python27\lib\site-packages\tox\session.py", line 375, in runcommand
    return self.subcommand_test()
  File "c:\python27\lib\site-packages\tox\session.py", line 548, in subcommand_test
    self.runtestenv(venv)
  File "c:\python27\lib\site-packages\tox\session.py", line 556, in runtestenv
    venv.test(redirect=redirect)
  File "c:\python27\lib\site-packages\tox\venv.py", line 332, in test
    ignore_ret=ignore_ret, testcommand=True)
  File "c:\python27\lib\site-packages\tox\venv.py", line 362, in _pcall
    redirect=redirect, ignore_ret=ignore_ret)
  File "c:\python27\lib\site-packages\tox\session.py", line 136, in popen
    stdout=stdout, stderr=STDOUT)
  File "c:\python27\lib\site-packages\tox\session.py", line 224, in _popen
    stdout=stdout, stderr=stderr, env=env)
  File "c:\python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "c:\python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified
```

<details><summary>Searchable metadata</summary>

```
trac-id__8578 8578
type__defect defect
reporter__rodrigc rodrigc
priority__normal normal
milestone__None None
branch__8578_rodrigc_tox_win32 8578-rodrigc-tox-win32
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468057760673177 1468057760673177
changetime__1468232316609926 1468232316609926
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

For those keeping track, the patch is here: https://github.com/twisted/twisted/pull/304/files

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_3"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Cool, I'm happy with this. Merge when ready.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset f9381660d3964e77e0b517ff1f227a389147669a
```
#!CommitTicketReference repository="" revision="f9381660d3964e77e0b517ff1f227a389147669a"
Merge pull request #304 from twisted/8578-rodrigc-tox-win32

Author: rodrigc
Reviewer: Lukasa
Fixes: #8578

Fix invoking tox on Windows to run tests
```
